services/nginz/third_party/nginx-zauth-module/zauth_module.c

```diff
@@ -495,21 +497,29 @@ static ngx_int_t zauth_parse_request (ngx_http_request_t * r) {
 }
 
 static ZauthResult token_from_header (ngx_str_t const * hdr, ZauthToken ** t) {
-        if (strncmp((char const *) hdr->data, "Bearer ", 7) == 0) {
-                return zauth_token_parse(&hdr->data[7], hdr->len - 7, t);
+        char const * bearer = "Bearer ";
+        size_t bearer_len = strlen(bearer);
+        char const * aws4_hmac_sha256 = "AWS4-HMAC-SHA256 ";
+        size_t aws4_hmac_sha256_len = strlen(bearer);
+        if (strncmp((char const *) hdr->data, bearer, bearer_len) == 0) {
```

Comment:
how do you know `hdr-> data` is non-NULL and `>= bearer_len`?  (same for elseif clause)